testdroid/__init__.py

```diff
@@ -160,7 +172,10 @@ def get_token(self):
     """ Helper method for getting necessary headers to use for API calls, including authentication
     """
     def _build_headers(self):
-        return { "Authorization": "Bearer %s" % self.get_token(), "Accept": "application/json" }
+        if not self.api_key:
```

Comment:
Would be cleaner and clearer without negation as `if self.api_key:... else: ...`